### PR TITLE
refs #17448 - renaming capsule to foreman-proxy

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

File: config/capsule.migrations/20161121212031_move_capsule_to_foreman_proxy_content.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# migrate answers from capsule if exist
+answers['capsule'].each do |k|
+  if k.eql? 'capsule'
+    answers['foreman_proxy_content']['foreman_proxy_content'] = answers['capsule'].delete k
```
Comment (stbenjam) [inline config/capsule.migrations/20161121212031_move_capsule_to_foreman_proxy_content.rb]: What's this supposed to be doing? A user will have an answers YAML like this:

```yaml
---
capsule:
  qpid_router_logging_path: 'whatever'
```

We just need that to become:

```yaml
---
foreman_proxy_content:
    qpid_router_logging_path: 'whatever'
```

So I think you can just do this or something similar:

```ruby
answers['foreman_proxy_content'] = answers.delete('capsule') if answers.key?('capsule')
```

None of the option names in the class itself have the word capsule in them:
https://github.com/Katello/puppet-capsule/blob/master/manifests/init.pp#L65-L91

File: hooks/post/10-post_install.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 def proxy_available?
-  Kafo::Helpers.module_enabled?(@kafo, 'capsule') &&
-   (@kafo.param('capsule', 'puppet').value ||
+  Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy' &&
```
Comment (stbenjam) [inline hooks/post/10-post_install.rb]: This is referring to the puppet module, which should be `foreman_proxy_content`

File: hooks/post/10-post_install.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+   (@kafo.param('foreman_proxy', 'puppet').value ||
```
Comment (stbenjam) [inline hooks/post/10-post_install.rb]: This is referring to the puppet module, which should be foreman_proxy_content

File: hooks/post/10-post_install.rb [hunk 4]
```diff
@@ -61,21 +61,27 @@ def new_install?
   Once this is completed run the steps below to start the smartproxy installation:
 
   1. Ensure that the foreman-installer-katello package is installed on the system.
+<<<<<<< 0cf41fa86030aa62562a6e87d7e33156038f9edb
```
Comment (stbenjam) [inline hooks/post/10-post_install.rb]: Unresolved merge conflict?

File: hooks/post/10-post_install.rb [hunk 5]
```diff
@@ -61,21 +61,27 @@ def new_install?
...
+=======
+  2. Copy <%= color("#{certs_tar}", :info) %> to the system <%= color("#{foreman_proxy_fqdn}", :info) %>
+  3. Run the following commands on the foreman-proxy (possibly with the customized
+     parameters, see <%= color("#{installer_name} --scenario foreman-proxy --help", :info) %> and
+>>>>>>> [WIP] Renaming capsule to foreman-proxy
      documentation for more info on setting up additional services):
 
-  #{installer_name} --scenario capsule\\
-                    --capsule-parent-fqdn                         "<%= "#{fqdn}" %>"\\
+  #{installer_name} --scenario foreman-proxy\\
```
Comment (stbenjam) [inline hooks/post/10-post_install.rb]: foreman-proxy-content

File: hooks/pre_migrations/10-migrate_legacy_foreman_proxy_config.rb [hunk 6]
```diff
@@ -1,6 +1,6 @@
-LEGACY_CONF_DIR = '/etc/capsule-installer'
-LEGACY_CONFIG_NAME = 'capsule-installer.yaml'
-SCENARIO_NAME = 'capsule.yaml'
+LEGACY_CONF_DIR = '/etc/foreman-proxy-installer'
+LEGACY_CONFIG_NAME = 'foreman-proxy-installer.yaml'
+SCENARIO_NAME = 'foreman-proxy.yaml'
```
Comment (stbenjam) [inline hooks/pre_migrations/10-migrate_legacy_foreman_proxy_config.rb]: Don't change old migrations, the original would still be correct

File: hooks/pre_validations/10-check_foreman_proxy_pulp.rb [hunk 7]
```diff
@@ -4,7 +4,7 @@ def error(message)
   kafo.class.exit 101
 end
 
-if param('capsule', 'pulp') && param('capsule', 'pulp').value
+if param('foreman-proxy', 'pulp') && param('foreman-proxy', 'pulp').value
```
Comment (stbenjam) [inline hooks/pre_validations/10-check_foreman_proxy_pulp.rb]: foreman_proxy_content

File: README.md [hunk 8]
```diff
@@ -38,26 +38,26 @@ foreman-installer --scenario                     "katello"\
...
                   --foreman-proxy-dns-zone       "example.com"\
-                  --capsule-puppet               "false"\
+                  --foreman-proxy-puppet         "false"\
                   --foreman-proxy-puppetca       "false"
 
-# Generate certificates for installing capsule on another system
-capsule-certs-generate --capsule-fqdn "mycapsule.example.com"\
-                       --certs-tar    "~/mycapsule.example.com-certs.tar"
+# Generate certificates for installing foreman-proxy on another system
+foreman-proxy-certs-generate --foreman-proxy-fqdn "myforeman-proxy.example.com"\
```
Comment (cfouant) [inline README.md]: Yes, understood, thanks.

File: config/capsule.migrations/20161121212031_move_capsule_to_foreman_proxy_content.rb (hunk 1 above)
Comment (cfouant) [inline config/capsule.migrations/20161121212031_move_capsule_to_foreman_proxy_content.rb]: Ah, okay, thanks. I was thinking there was a capsule option inside the class. Thanks!

File: hooks/pre_migrations/10-migrate_legacy_foreman_proxy_config.rb (hunk 6 above)
Comment (cfouant) [inline hooks/pre_migrations/10-migrate_legacy_foreman_proxy_config.rb]: ah missed turning this one back. thanks!

File: README.md [hunk 9]
```diff
@@ -27,7 +27,7 @@ foreman-installer --scenario                     "katello"\
                   --foreman-proxy-dhcp           "true"\
                   --foreman-proxy-dhcp-interface "virbr1"\
                   --foreman-proxy-tftp           "true"\
-                  --capsule-puppet               "true"\
+                  --foreman-proxy-puppet         "true"\
```
Comment (stbenjam) [inline README.md]: This one too, should be `--foreman-proxy-content-puppet`

File: hooks/post/10-post_install.rb [hunk 10]
```diff
@@ -1,10 +1,10 @@
 def proxy_available?
-  Kafo::Helpers.module_enabled?(@kafo, 'capsule') &&
-   (@kafo.param('capsule', 'puppet').value ||
-    @kafo.param('foreman_proxy', 'puppetca').value ||
-    @kafo.param('foreman_proxy', 'dhcp').value ||
-    @kafo.param('foreman_proxy', 'dns').value ||
-    @kafo.param('foreman_proxy', 'tftp').value)
+  Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy' &&
```
Comment (stbenjam) [inline hooks/post/10-post_install.rb]: This one too

File: hooks/post/10-post_install.rb [hunk 11]
```diff
@@ -35,16 +35,16 @@ def new_install?
...
 
 MSG
     end
 
-    if Kafo::Helpers.module_enabled?(@kafo, 'capsule_certs')
-      if certs_tar = @kafo.param('capsule_certs', 'certs_tar').value
-        capsule_fqdn          = @kafo.param('capsule_certs', 'capsule_fqdn').value
+    if Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_certs')
+      if certs_tar = @kafo.param('foreman_proxy_certs', 'certs_tar').value
+        foreman_proxy_fqdn          = @kafo.param('foreman_proxy_certs', 'foreman_proxy_fqdn').value
```
Comment (stbenjam) [inline hooks/post/10-post_install.rb]: Nitpick, can you realign the ='s spacing to match the ones below (Or remove the alignment altogother)?

File: spec/migration_spec.rb [hunk 12]
```diff
@@ -6,7 +6,7 @@
 Kafo::KafoConfigure.logger = Logger.new("test.log")
 
 describe 'migrations' do
-  %w(katello katello-devel capsule).each do |scenario_name|
+  %w(katello katello-devel foreman-proxy).each do |scenario_name|
```
Comment (stbenjam) [inline spec/migration_spec.rb]: foreman-proxy-content (scenario name)

**Comment by stbenjam** [comment]
Looks pretty good, just a few small things.  We should be in good shape to test this all together once pipeline is green again

**Comment by cfouant** [comment]
@stbenjam - not sure if we are renaming the puppet-capsule repo to puppet-foreman-proxy-content. In order to get some green tests, I've left the Puppetfile calling git for puppet-capsule at this time. Let me know if that will need updating!

**Comment by stbenjam** [comment]
We will, once it's merged.  I think maybe you can do this to point at my branch

```
mod 'katello/foreman_proxy_content', :git => 'https://github.com/stbenjam/puppet-capsule', :ref => '17366'
```

**Comment by stbenjam** [comment]
Latest update looks good to me! I don't see anything missing.  Hopefully we can actually try it soon.

**Comment by stbenjam** [comment]
@cfouant Can you rebase your PR branch?

Need that so this fix is included so I can test the installer:
  https://github.com/Katello/katello-installer/commit/c01ce6945a1802cbeb3676def48f22aaf567f564

**Comment by cfouant** [comment]
@stbenjam - done!

**Comment by stbenjam** [comment]
I get a good katello install!

```
[root@bye-capsule ~]# foreman-installer --scenario katello --disable-system-checks
Installing             Done                                               [100%] [...........................................................................................................................................................]
  Success!
  * Katello is running at https://bye-capsule.example.com
      Initial credentials are admin / k5a6W29gxjNRvEPs
  * To install an additional Foreman proxy on separate machine continue by running:

      foreman-proxy-certs-generate --foreman-proxy-fqdn "$FOREMAN-PROXY" --certs-tar "~/$FOREMAN-PROXY-certs.tar"

  The full log is at /var/log/foreman-installer/katello.log
```

My box is:

```
bye-capsule:
  box: centos7
  memory: 6144
  ansible:
    playbook: 'playbooks/katello.yml'
    group: 'server'
    variables:
        katello_version: nightly
        foreman_version: nightly
        puppet_version: 4
        foreman_installer_skip_installer: True
        foreman_repositories_use_koji: True
        katello_repositories_use_koji: True
        puppet_repositories_version: 4
        foreman_installer_module_prs: "katello/capsule/106,katello/certs/111"
```

And you'll need this script here to apply the changes from this PR:
  https://gist.github.com/stbenjam/0995d0c02c72554e94d28641c793f379

**Comment by stbenjam** [comment]
Next up is testing an upgrade and a capsule.

**Comment by stbenjam** [comment]
@cfouant In this file:
 https://github.com/Katello/katello-installer/blob/master/hooks/pre/17-memory_check.rb

We need a check for 'foreman_proxy_certs' instead of 'capsule_certs' now

**Comment by stbenjam** [comment]
Another note, once https://github.com/Katello/katello-installer/pull/446 is merged, we'll need to rebase and get those changes applied to foreman-proxy-content.yaml too.

**Comment by cfouant** [comment]
@stbenjam - changes made, rebased as well :)

**Comment by stbenjam** [comment]
So an issue I encountered during the upgrade - we need to do a few things to migrate a "capsule" to use a "foreman-proxy-content" scenario.  Can you move  the entire capsule.migrations directory in the installer should now be foreman-proxy-content.migrations?

Then in the RPM we need to:
- [ ] Move capsule.migrations/.applied to foreman-proxy-content.migrations/ so we don't re-apply applied migrations
- [ ] Remove the symlink from last_scenario.yaml to capsule.yaml and replace it with foreman-proxy-content.yaml

I can take care of the RPM things if you want to handle changing the capsule.migrations dir here

**Comment by stbenjam** [comment]
But after the above I'm able to "upgrade" a capsule to be a "foreman proxy with content"

**Comment by stbenjam** [comment]
Required PR's for everything to be merged:
- [ ] This PR
- [ ] https://github.com/Katello/puppet-certs/pull/111
- [ ] https://github.com/Katello/puppet-capsule/pull/106
- [ ] https://github.com/Katello/forklift/pull/338

**Comment by ehelms** [comment]
I tried to test the box from above doing:

 * vagrant up bye-capsule
 * ran installer_pr.sh
 * ran foreman-installer --scenario katello

I hit this on the installer run part:

```
[ERROR 2016-12-13 02:23:22 main] ^[[1;31mError: Evaluation Error: Error while evaluating a Function Call, Could not find class ::foreman_proxy_content::params for bye-capsule.example.com  at line 1:365 on node bye-capsule.example.com^[[0m
```

**Comment by stbenjam** [comment]
Tried it again and it's installing fine, no errors like that. Do you have more info? Did you use the forklift PR?

**Comment by ehelms** [comment]
For some reason, (I have not dug into why) the capsule PR was not getting checked out. Manually performed that step and it's chugging along now.

**Comment by stbenjam** [comment]
> For some reason, (I have not dug into why) the capsule PR was not getting checked out. Manually performed that step and it's chugging along now.

That's odd, do you have the latest forklift with the module pr testing bits?

**Comment by ehelms** [comment]
I used your Forklift PR for this work.

In other news, I had happy installs! Also, in testing this, I found a Pulp bug. :+1:

**Comment by stbenjam** [comment]
Great! We can merge once the first item in https://github.com/Katello/katello-installer/pull/437#issuecomment-265834916 is addressed, that impacts capsule upgrades.  Just a simple rename of the directory

**Comment by cfouant** [comment]
@ehelms @stbenjam - I think we're done? The test failures don't seem to be related to my PR but I'm looking into it!

**Pull request merged**